flink-core/src/main/java/org/apache/flink/core/memory/MemorySegment.java

```diff
@@ -1349,4 +1377,38 @@ else if (seg2.address > seg2.addressLimit) {
 					String.format("offset1=%d, offset2=%d, len=%d, bufferSize=%d, address1=%d, address2=%d",
 							offset1, offset2, len, tempBuffer.length, this.address, seg2.address));
 	}
+
+	/**
+	 * Equals two memory segment regions.
+	 *
+	 * @param seg2 Segment to equal this segment with
+	 * @param offset1 Offset of this segment to start equaling
+	 * @param offset2 Offset of seg2 to start equaling
+	 * @param length Length of the equaled memory region
+	 *
+	 * @return true if equal, false otherwise
+	 */
+	public final boolean equalTo(MemorySegment seg2, int offset1, int offset2, int length) {
```

Comment:
let `offset1` be the first parameter will make this looks more compatible with other methods?